tests/vmi_multus_test.go

```diff
@@ -1008,16 +1042,75 @@ var _ = Describe("[Serial]Macvtap", func() {
 
 		BeforeEach(func() {
 			macAddress := "02:03:04:05:06:07"
-			macvtapVMI = createCirrosVMIWithMacvtapDefinedMAC(virtClient, macvtapNetworkName, macAddress)
+			clientVMI = createCirrosVMIStaticIPRandomNode(virtClient, macvtapNetworkName, macAddress)
 		})
 
 		It("should be successful when the VMI MAC address is defined in its spec", func() {
 			By("starting the migration")
-			migration := tests.NewRandomMigration(macvtapVMI.GetName(), macvtapVMI.GetNamespace())
+			migration := tests.NewRandomMigration(clientVMI.GetName(), clientVMI.GetNamespace())
 			migrationUID := tests.RunMigrationAndExpectCompletion(virtClient, migration, migrationWaitTime)
 
 			// check VMI, confirm migration state
-			tests.ConfirmVMIPostMigration(virtClient, macvtapVMI, migrationUID)
+			tests.ConfirmVMIPostMigration(virtClient, clientVMI, migrationUID)
+		})
+
+		Context("with live traffic", func() {
+			var serverVMI *v1.VirtualMachineInstance
+			var serverIP string
+
+			getVMMacvtapIfaceIP := func(vmName string, namespace string, macAddress string) (string, error) {
+				var vmiIP string
+				err := wait.PollImmediate(time.Second, 2*time.Minute, func() (done bool, err error) {
+					vmi, err := virtClient.VirtualMachineInstance(namespace).Get(vmName, &v13.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+
+					for _, iface := range vmi.Status.Interfaces {
+						if iface.MAC == macAddress {
+							vmiIP = iface.IP
+							return true, nil
+						}
+					}
+
+					return false, nil
+				})
+				if err != nil {
+					return "", err
+				}
+
+				// code below should be removed if / when KubeVirt honors its
+				// API and returns an IP on the Interface.IP attribute
+				// Track the following PR to know when to update:
+				//   - https://github.com/kubevirt/kubevirt/pull/4217
+				ipAndNetworkLength := strings.Split(vmiIP, "/")
+				if len(ipAndNetworkLength) != 2 {
+					return "", fmt.Errorf("could not compute the IP address from the CIDR")
+				}
+				return ipAndNetworkLength[0], err
```

Comment:
Nit, won't fight for it: Would you return nil instead of err. Although err happens to be nil here, it is just a "side effect" taken from the poll.